src/main/kotlin/org/rust/cargo/runconfig/buildtool/CargoBuildEventsConverter.kt

```diff
@@ -103,8 +82,8 @@ class CargoBuildEventsConverter(private val context: CargoBuildContext) : BuildO
         return true
     }
 
-    private fun tryHandleRustcArtifact(json: JsonObject?): Boolean {
-        val rustcArtifact = json?.let { CargoMetadata.Artifact.fromJson(it) } ?: return false
+    private fun tryHandleRustcArtifact(jsonObject: JsonObject): Boolean {
+        val rustcArtifact = CargoMetadata.Artifact.fromJson(jsonObject) ?: return false
```

Comment:
I've found endless building when compiling some small Rust project. Looks like `JsonObject` was incomplete and `json.getAsJsonPrimitive("reason")` didn't finish successfully. At least `json.getAsJsonPrimitive("reason")` should be checked for null value.